ui/src/dashboards/components/TemplateControlBar.tsx

```diff
@@ -0,0 +1,63 @@
+import React, {SFC} from 'react'
+
+import classnames from 'classnames'
+import uuid from 'uuid'
+
+import Authorized, {EDITOR_ROLE} from 'src/auth/Authorized'
+import TemplateControlDropdown from 'src/dashboards/components/TemplateControlDropdown'
+
+interface TemplateValue {
+  value: string
+  selected?: boolean
+}
+
+interface Template {
+  id: string
+  tempVar: string
+  values: TemplateValue[]
+}
+
+interface Props {
+  templates: Template[]
+  isOpen: boolean
+  onOpenTemplateManager: () => void
+  onSelectTemplate: (id: string) => void
+}
+
+const TemplateControlBar: SFC<Props> = ({
+  isOpen,
+  templates,
+  onSelectTemplate,
+  onOpenTemplateManager,
+}) => (
+  <div className={classnames('template-control-bar', {show: isOpen})}>
+    <div className="template-control--container">
+      <div className="template-control--controls">
+        {templates && templates.length ? (
```

Comment:
how come you added `templates &&` here? Is `nil` an allowed value for the `Template[]` type?

Reply:
sometimes the component gets templates as `undefined` so just guarding against that